vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpacesCustomOrderManager.java

```diff
@@ -0,0 +1,158 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.persistence.EntityNotFoundException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.asu.diging.vspace.core.data.SpacesCustomOrderRepository;
+import edu.asu.diging.vspace.core.model.IExhibition;
+import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.model.ISpacesCustomOrder;
+import edu.asu.diging.vspace.core.model.impl.Exhibition;
+import edu.asu.diging.vspace.core.model.impl.SpaceStatus;
+import edu.asu.diging.vspace.core.model.impl.SpacesCustomOrder;
+import edu.asu.diging.vspace.core.services.IExhibitionManager;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+import edu.asu.diging.vspace.core.services.ISpacesCustomOrderManager;
+
+/**
+ * SpacesCustomOrderManager is the manager
+ * to allow custom ordering of spaces
+ * @author Glen D'souza
+ *
+ */
+
+@Service
+@Transactional(rollbackFor = { Exception.class })
+public class SpacesCustomOrderManager implements ISpacesCustomOrderManager {
+    
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private SpacesCustomOrderRepository spacesCustomOrderRepository;
+    
+    @Autowired
+    private ISpaceManager spaceManager;
+    
+    @Autowired
+    private IExhibitionManager exhibitionManager;
+    
+    @Override
+    public ISpacesCustomOrder create(List<String> spaceOrders,
+            String name,
+            String description) {
+        SpacesCustomOrder spacesCustomOrder = new SpacesCustomOrder();
+        spacesCustomOrder.setCustomOrderName(name);
+        spacesCustomOrder.setDescription(description);
+        List<ISpace> orderedSpaces = new ArrayList<ISpace>();        
+        orderedSpaces = spaceOrders.stream()
+                .map(spaceId -> spaceManager.getSpace(spaceId))
+                .collect(Collectors.toList());
+        spacesCustomOrder.setCustomOrderedSpaces(orderedSpaces);
+        return spacesCustomOrderRepository.save(spacesCustomOrder);
+    }
+
+    
+    @Override
+    public Iterable<SpacesCustomOrder> findAll(){
+        return spacesCustomOrderRepository.findAll();
+    }
+    
+    @Override
+    public ISpacesCustomOrder get(String customSpaceOrderId) {
+        try {
+            Optional<SpacesCustomOrder> spacesCustomOrderOptional = spacesCustomOrderRepository.findById(customSpaceOrderId);
+            return spacesCustomOrderOptional.get();
+        } catch (NoSuchElementException e) {
```

Comment:
exception should only be thrown if it can't be anticipated for. Just check if the optional has an element and only call `get` if it does.